src/redux/reducer.js

```diff
@@ -0,0 +1,107 @@
+import { combineReducers } from 'redux';
```

Comment:
Just a thought, we can use reducers to do some of the transformation (like your building of the cart in the cart file) here, so that it only happens when an action fires, and not per re-render.